### PR TITLE
Renamed compiled library to `libidwarp`

### LINKED ISSUE #24: Rename the compiled `.so` libraries
## Type of issue
What types of issue is it?
_Select the appropriate type(s) that describe this issue_

- Maintenance

## Description
In order to reduce confusion with the Python package, we should rename the compiled libraries by appending `lib` in front. This is consistent with the other repositories, such as ADflow.
- `idwarp.so` -> `libidwarp.so`
- `idwarp_cs.so` -> `libidwarp_cs.so`

In the current situation, when typing `import idwarp` it is unclear if we mean the compiled module, or the Python package.

Both the Python script and the Makefiles need to be updated for this to work properly.

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
Closes #24.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
Build/tests work locally for both real and complex.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,8 +20,8 @@ src/warp/temp_reverse/*
...
 *.plt
 *.html
-src_cs/f2py/warpustruct.pyf.autogen
-src/f2py/warpustruct.pyf.autogen
-src/f2py/warpustruct-f2pywrappers2.f90
-src_cs/f2py/warpustruct_cs-f2pywrappers2.f90
+src_cs/f2py/idwarp.pyf.autogen
+src/f2py/idwarp.pyf.autogen
+src/f2py/idwarp-f2pywrappers.f90
+src_cs/f2py/idwarp_cs-f2pywrappers.f90
```
Comment (eirikurj) [inline .gitignore:26]: Naming here is inconsistent with the actual filename, should be `libidwarp`

File: .gitignore [hunk 2]
```diff
@@ -20,8 +20,8 @@ src/warp/temp_reverse/*
 *.lay
 *.plt
 *.html
-src_cs/f2py/warpustruct.pyf.autogen
-src/f2py/warpustruct.pyf.autogen
-src/f2py/warpustruct-f2pywrappers2.f90
-src_cs/f2py/warpustruct_cs-f2pywrappers2.f90
+src_cs/f2py/idwarp.pyf.autogen
+src/f2py/idwarp.pyf.autogen
```
Comment (eirikurj) [inline .gitignore:24]: I would actually also rename the pyf file to `libidwarp` since the pyf file is for the module.

**Review by eirikurj — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (nwu63) [reply .gitignore:26]: Whoops good catch

File: .gitignore (hunk 2 above)
Comment (nwu63) [reply .gitignore:24]: Right that makes sense. I didn't do it since adflow didn't do it but I'll make the changes here

**Review by eirikurj — APPROVED** [review]

**Review by Xiaosong2105 — APPROVED** [review]

**Pull request merged**